File: javascript/228.summary-ranges.js
/*
 * @lc app=leetcode id=228 lang=javascript
 *
 * [228] Summary Ranges
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function (nums) {
  const ranges = [];
  let slow = 0;
  for (let fast = 1; fast <= nums.length; fast++) {
    if (fast === nums.length || nums[fast - 1] + 1 !== nums[fast]) {
      let range = `${nums[slow]}`;
      if (slow !== fast - 1) {
        range += `->${nums[fast - 1]}`;
      }
      ranges.push(range);
      slow = fast;
    }
  }
  return ranges;
};
// @lc code=end

File: rust/228.summary-ranges.rs
/*
 * @lc app=leetcode id=228 lang=rust
 *
 * [228] Summary Ranges
 */

// @lc code=start
impl Solution {
    pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
        let mut result: Vec<String> = Vec::new();
        let mut slow = 0;
        for fast in 1..=nums.len() {
            if fast == nums.len() || nums[fast] - nums[fast - 1] != 1 {
                let mut range = String::from(nums[slow].to_string());
                if fast - 1 != slow {
                    range.push_str(format!("->{}", nums[fast - 1]).as_ref());
                }
                result.push(range);
                slow = fast;
            }
        }
        return result;
    }
}
// @lc code=end
